packages/capabilities/test/capabilities/usage.test.js

```diff
@@ -200,4 +203,47 @@ describe('usage capabilities', function () {
       })
     }, /Expected value of type integer instead got 6\.6/)
   })
+
+  it('should delegate and invoke usage/record', async () => {
+    const data = {
+      space: mallory.did(),
+      resource: readmeCID,
+      bytes: 100,
+      servedAt: 1714204800,
+    }
+
+    // W3 delegates ability to record usage to Gateway
+    const usageRecordDelegationProof = await Usage.record.delegate({
+      issuer: w3,
+      audience: gateway,
+      with: w3.did(),
+      expiration: Infinity,
+    })
+
+    // Gateway invokes usage/record and indicates the w3 as the audience
+    const recordInvocation = Usage.record.invoke({
+      issuer: gateway,
+      audience: w3,
+      with: gateway.did(),
+      nb: { ...data },
+      proofs: [usageRecordDelegationProof],
+    })
+
+    // W3 validates the delegation from Gateway to itself
+    const result = await access(await recordInvocation.delegate(), {
+      capability: Usage.record,
+      principal: Verifier,
+      authority: w3,
+      validateAuthorization,
+      resolveDIDKey: () => Schema.ok(gateway.toDIDKey()),
```

Comment:
question: is this going to resolve all `did:web`s to the gateway key? I'm concerned this might hide a test failure by thinking that the `w3` DID uses the same private key and is therefore the same entity as the `gateway` principal - maybe misunderstanding but curious what you think... it does look like `resolveDIDKey` is passed a `DID` - maybe it can return different DIDKeys for different DIDs with a lookup table here?

Reply:
ok so this is what I'm imagining:

```suggestion
      resolveDIDKey: (k) => {
        const didKey = [w3, gateway].find(signer => signer.did() === k)?.toDIDKey()
        if (didKey) {
          return Schema.ok(didKey)
        } else {
          return { error: new DIDResolutionError(k) }
        }
      },
```